### PR TITLE
feat: prevent sending in ChatInput while IME conversion is in progress

### LINKED ISSUE #109: [Feat]Disable Enter key events during IME conversion in ChatInput
**Is your feature request related to a problem? Please describe.**
The problem occurs in the ChatInput component. Currently, when the Enter key is pressed while text is undergoing IME (Input Method Editor) conversion, it triggers form submission events. This can lead to incomplete or unintended input submissions, which is particularly frustrating when composing messages in languages that require conversion, like Japanese.

**Describe the solution you'd like**
I want to add an isComposing state to the ChatInput component to prevent events from being triggered when the Enter key is pressed during IME conversion

**Describe alternatives you've considered**

**Additional context**

https://github.com/user-attachments/assets/f0c352d8-0fbe-4f7c-be09-79a8de419b97

### PR DESCRIPTION
**Pull request by daichi000**
#### Description

Add an isComposing state to the ChatInput component to prevent events from being triggered when the Enter key is pressed during IME conversion

Closes #109 

#### What change does this PR introduce?

Please select the relevant option(s).

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (changes to docs/code comments)

#### What is the proposed approach?

Please give a short overview/explanation on the approach taken to resolve the issue(s).

#### Checklist:

- [x] The commit message follows our adopted [guidelines](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] Testing has been done for the change(s) added (for bug fixes/features)
- [ ] Relevant comments/docs have been added/updated (for bug fixes/features)

### CODE REVIEW

File: src/components/ChatBotInput/ChatBotInput.tsx [hunk 1]
```diff
@@ -103,12 +106,27 @@ const ChatBotInput = ({ buttons }: { buttons: JSX.Element[] }) => {
...
+		setIsComposing(false);
+	};
+
 	/**
 	 * Handles keyboard events and proceeds to submit user input if enter button is pressed.
 	 * 
 	 * @param event keyboard event
 	 */ 
 	const handleKeyDown = async (event: KeyboardEvent<HTMLTextAreaElement | HTMLInputElement>) => {
+		if (isComposing) return;
```
Comment (tjtanjin) [inline src/components/ChatBotInput/ChatBotInput.tsx:129]: Let's avoid one-liners and use egyptian braces instead.

**Review by tjtanjin — CHANGES_REQUESTED** [review]

File: src/components/ChatBotInput/ChatBotInput.tsx (hunk 1 above)
Comment (daichi000) [reply src/components/ChatBotInput/ChatBotInput.tsx]: Fixed.
[chore: use egyptian braces](https://github.com/tjtanjin/react-chatbotify/pull/112/commits/1bcc299688bdb0f2ba908f00e475a653a10cc217)

**Comment by daichi000** [comment]
I wanted to add tests, but skipped them because Selenium does not support IME conversion...
I have confirmed that all existing tests are passing.

**Comment by tjtanjin** [comment]
> I wanted to add tests, but skipped them because Selenium does not support IME conversion...
> 
> I have confirmed that all existing tests are passing.

Thank you for checking, there's an existing task to move to cypress haha - not sure if that would help with this though 😝

**Pull request merged**